### PR TITLE
Add functional tests + fix grants order

### PR DESCRIPTION
**Pull request by BibianaC**
This PR was originally to add functional tests but ended up being a fix of issue #40 (grants' order was incorrect).

It seems a long PR but it is not:
- registry/registry.py & registry/views.py -> Is where the fix is. I have also refactored a little bit.
- tests/samples/registry_raw_data.py -> It is only an update of the data.
- tests/test_functional.py _> Only 'test_table' are new, the rest are just commented out. There are many missing tests but I didn't want to make this PR bigger. I will add them in another PR.

### CODE REVIEW

**Comment by BibianaC** [comment]
Thanks @robredpath . Let's see what they say.

File: registry/registry.py [hunk 1]
```diff
@@ -201,11 +201,32 @@ def get_data_by_prefix(raw_data):
 
         data_by_prefix[prefix]['grant'].append(get_grant_data(data))
 
-    return OrderedDict(sorted(data_by_prefix.items(), key=lambda x: x[1]['publisher']['name']))
+    return data_by_prefix
+
+
+def sort_data_by_prefix(data_by_prefix):
+    """
+    Sort grants by the year of the first date and publishers by their name.
```
Comment (robredpath) [inline registry/registry.py]: I think this comment is misleading, as the *result* of the sorting is that the list is ordered first by publisher name, and then by date of first grant, even if the mechanism of doing it is to sort by date first.

File: registry/registry.py [hunk 2]
```diff
@@ -201,11 +201,32 @@ def get_data_by_prefix(raw_data):
 
         data_by_prefix[prefix]['grant'].append(get_grant_data(data))
 
-    return OrderedDict(sorted(data_by_prefix.items(), key=lambda x: x[1]['publisher']['name']))
+    return data_by_prefix
+
+
+def sort_data_by_prefix(data_by_prefix):
```
Comment (robredpath) [inline registry/registry.py]: I think this function name is misleading, as we're using publisher name and date to sort by, not prefix.

File: registry/registry.py [hunk 3]
```diff
@@ -201,11 +201,32 @@ def get_data_by_prefix(raw_data):
...
 
 
 def get_raw_data():
     if os.environ.get('FLASK_ENV') == 'development':
         return RAW_DATA
 
     return requests.get('https://storage.googleapis.com/datagetter-360giving-output/branch/master/status.json').json()
+
+
+def registry_main():
```
Comment (robredpath) [inline registry/registry.py]: This seems like an odd name, to me - like, I guess it's a reference to `__main__`, but it's also....not that. 

Looking at how it's used, it seems to be about getting data, sorted by prefix. Would `get_data_sorted_by_prefix()` be an appropriate name?

File: registry/registry.py (hunk 1 above)
Comment (BibianaC) [reply registry/registry.py]: I have edited the line.

File: registry/registry.py (hunk 3 above)
Comment (BibianaC) [reply registry/registry.py]: This is a standard way of calling it in other companies I worked. The proposed name is fine too. I have changed it.

**Pull request merged**